File: fixtures/extraction/resolution_contract/csharp/static_type_receiver/fixture.cs
// Static-type receiver for C#: the receiver names a type directly rather than a
// variable whose type must be inferred, so no `type_facts` row participates.
// `Fixture.Create()` and `Color.Red` both resolve at tier3_static_type (0.70)
// from another file, because `Fixture` and `Color` are public and top-level.
namespace App
{
    public class Fixture
    {
        public static int Create() { return 1; }
    }

    public enum Color
    {
        Red,
        Blue
    }

    public class Limits
    {
        public const int Max = 10;
    }
}

File: fixtures/extraction/resolution_contract/typescript/static_type_receiver/fixture.ts
// Same-file static-type receiver: no import path, so tier2 cannot bind.
// Fixture.create() resolves via tier3_static_type.
export class Fixture {
  static create(): number {
    return 1;
  }
}

export class Limits {
  static max(): number {
    return 10;
  }
}

export function run(): number {
  const made = Fixture.create();
  return made + Limits.max();
}
